### PR TITLE
Updates for week of July 22

### PR DESCRIPTION
**Pull request by mbolivar**
- pull in mergeups and atomically keep applications compiling (for bisectability)
- convert deprecated device tree macros to new names in follow-up commits
- fix other build warnings observed
- rename west projects to logical names

Mergeups:

https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/pull/47
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/179

Related nrfxlib PR:

https://github.com/NordicPlayground/nrfxlib/pull/78

### CODE REVIEW

**Comment by mbolivar** [comment]
CC: @mike-scott

File: samples/nrf9160/gps/src/main.c [hunk 1]
```diff
@@ -37,12 +37,12 @@ nrf_gnss_data_frame_t last_fix;
 
 void bsd_recoverable_error_handler(uint32_t error)
 {
-	printf("Err: %lu\n", error);
+	printf("Err: %u\n", error);
```
Comment (rlubos) [inline samples/nrf9160/gps/src/main.c]: This actually depends on what standard library is used. Zephyrs minimal libc indeed defines `uint32_t` as `unsigned int`, yet the newlib defines it as `unsigned long`, and the sample appears to use newlib. Perhaps we should use `"PRIu32"` to be on a safe side?

Comment (mbolivar) [reply samples/nrf9160/gps/src/main.c]: > Perhaps we should use "PRIu32" to be on a safe side?

I would rather cast to unsigned long in the printf statement if that is the case -- OK for you?

Comment (rlubos) [reply samples/nrf9160/gps/src/main.c]: Sure

**Comment by mbolivar** [comment]
@thst-nordic this is the error:

```
  The SDK version you are using is too old, please update your SDK.


  You need at least SDK version 0.10.1.


  You have version 0.10.0 (/opt/zephyr-sdk).


  The new version of the SDK can be downloaded from:
```

I was able to find the docker.yaml file in the CI_LIB repository, but not the repository containing the Dockerfile itself which would let me submit a PR updating the zephyr SDK installed. Any pointers?

**Comment by mbolivar** [comment]
I've merged upstream again into this PR.

**Comment by mbolivar** [comment]
Dependent PRs are merged, west.yml updated. Good to go with a +1.

**Pull request merged**